### PR TITLE
Fix and refactor

### PR DESCRIPTION
**Pull request by robinbernon**
Fixing all compiler errors and warnings. Should now be able to clone this repo and run "make test" without any issues. For this to work a change was required to the substreams client: https://github.com/streamingfast/substreams/pull/53.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ jobs:
       - name: install substreams cli
         uses: jaxxstorm/action-install-gh-release@v1.7.1
         with:
-          repo: streamingfast/substreams
+          repo: robinbernon/substreams-messari
```
Comment (this-username-is-taken) [inline .github/workflows/ci.yaml:17]: why `robinbernon/substreams-messari`?

File: Makefile [hunk 2]
```diff
@@ -8,6 +8,14 @@ build-all:
 	$(MAKE) -C erc721 build
 	$(MAKE) -C compound-v2 build
 
-.PHONY: erc20-price-spkg
-erc20-price-spkg:
-	cd erc20-price && make pack && mv erc20-price-substreams-v0.1.0.spkg ../target
\ No newline at end of file
+.PHONY: example-all
```
Comment (this-username-is-taken) [inline Makefile:11]: What are your thoughts on changing this to `run-all` and changing `example` to `run`? I feel like the latter is more self-explanatory

**Review by this-username-is-taken — DISMISSED** [review]
Great work! Added a couple minor comments otherwise LGTM.

File: Makefile (hunk 2 above)
Comment (robinbernon) [reply Makefile:11]: Yep sounds good! - Would you like me to change all individual `make example` commands to `make run` also?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (robinbernon) [reply .github/workflows/ci.yaml]: They have now merge my PR into their main branch so I've put this back to what it was originally. Unfortunately they have not yet made a release for the new commit to master so the build agent is still pulling down the previous version which is why build is still failing.I'll keep an eye out for when they make the new release.

File: Makefile (hunk 2 above)
Comment (this-username-is-taken) [reply Makefile:11]: yes please

**Comment by this-username-is-taken** [comment]
btw, we have some rough guidelines around PR naming convention for the subgraph repo: https://github.com/messari/subgraphs/blob/master/docs/Contributing.md#naming-conventions.

Doesn't 100% applies here but you can use it (and the previous PRs' name) as a reference.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (this-username-is-taken) [reply .github/workflows/ci.yaml]: sounds good!

**Review by this-username-is-taken — APPROVED** [review]
Great work!

**Pull request merged**